easy_phi/hwconf.py

```diff
@@ -0,0 +1,91 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+"""This module provides functions to obtain hardware configuration and keep it up to date.
+
+"""
+
+import pyudev
+import hwal
+from tornado.options import options
+
+callbacks = [
+    # hardware configuration change listener will call these methods
+    # it is necessary to send updates to websockets
+]
+
+# device #0 represents broadcast
+modules = [None for i in range(options.slots + 1)]
+modules[0] = hwal.BroadcastModule(modules)
+
+_context = pyudev.Context()
+
+
+def get_rack_slot(device):
+    global modules  # Ugly, I know. Let me know if there is a better way
+
+    # first, check if device is already represented in modules
+    for i, module in enumerate(modules):
+        if module.device == device:
+            return i
+
+    # if it is not in the list, match it with USB ports of the rack
+    # TODO: get usb ports list from config file (command line option)
+    slot = 1
+
+    # if device is neither in modules list nor port is associated with rack slot, assign to first free slot
+    if not 1 <= slot <= options.slots:
+        for i in range(options.slots + 1, len(modules)):
+            if modules[i] is None:
+                return i
+        slot = len(modules)
+        modules += [None]
+    return slot
+
+
+def hwconf_update():
+    """Check connected devices and refresh modules list.
+    Usually this method is called upon system startup or restart to
+    restore current hardware configuration
+    """
+    for device in _context.list_devices():
+        for module_class in hwal.__all__:
```

Comment:
I would like to hear the reasoning why naming an array "module_classes" is bad and unacceptable in this case. My reasoning behind "module_classes" is Maintainability quality attribute, as it would be easier to understand this file for a person not deeply familiar with python's language-specific constructs. If there is a counterargument agains naming "module_classes" or strong argument in favour of "__all__" - I'd be glad to accept your point of view